<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN"
  "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">


<html xmlns="http://www.w3.org/1999/xhtml">
  <head>
    <meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
    
    <title>Picking OTUs for use in PICRUSt &mdash; PICRUSt 1.1.3 documentation</title>
    
    <link rel="stylesheet" href="../_static/sphinxdoc.css" type="text/css" />
    <link rel="stylesheet" href="../_static/pygments.css" type="text/css" />
    
    <script type="text/javascript">
      var DOCUMENTATION_OPTIONS = {
        URL_ROOT:    '../',
        VERSION:     '1.1.3',
        COLLAPSE_INDEX: false,
        FILE_SUFFIX: '.html',
        HAS_SOURCE:  true
      };
    </script>
    <script type="text/javascript" src="../_static/jquery.js"></script>
    <script type="text/javascript" src="../_static/underscore.js"></script>
    <script type="text/javascript" src="../_static/doctools.js"></script>
    <link rel="top" title="PICRUSt 1.1.3 documentation" href="../index.html" /> 
  </head>
  <body role="document">
    <div class="related" role="navigation" aria-label="related navigation">
      <h3>Navigation</h3>
      <ul>
        <li class="right" style="margin-right: 10px">
          <a href="../genindex.html" title="General Index"
             accesskey="I">index</a></li>
        <li class="nav-item nav-item-0"><a href="../index.html">PICRUSt 1.1.3 documentation</a> &raquo;</li> 
      </ul>
    </div>
      <div class="sphinxsidebar" role="navigation" aria-label="main navigation">
        <div class="sphinxsidebarwrapper">
  <h3><a href="../index.html">Table Of Contents</a></h3>
  <ul>
<li><a class="reference internal" href="#">Picking OTUs for use in PICRUSt</a><ul>
<li><a class="reference internal" href="#introduction">Introduction</a></li>
<li><a class="reference internal" href="#picking-closed-reference-otus-with-qiime">Picking closed reference OTUs with QIIME</a></li>
<li><a class="reference internal" href="#using-an-open-reference-otu-table-from-qiime">Using an open reference OTU table from QIIME</a></li>
<li><a class="reference internal" href="#removing-samples-with-0-read-counts">Removing samples with 0 read counts</a></li>
</ul>
</li>
</ul>

  <div role="note" aria-label="source link">
    <h3>This Page</h3>
    <ul class="this-page-menu">
      <li><a href="../_sources/tutorials/otu_picking.txt"
            rel="nofollow">Show Source</a></li>
    </ul>
   </div>
<div id="searchbox" style="display: none" role="search">
  <h3>Quick search</h3>
    <form class="search" action="../search.html" method="get">
      <input type="text" name="q" />
      <input type="submit" value="Go" />
      <input type="hidden" name="check_keywords" value="yes" />
      <input type="hidden" name="area" value="default" />
    </form>
    <p class="searchtip" style="font-size: 90%">
    Enter search terms or a module, class or function name.
    </p>
</div>
<script type="text/javascript">$('#searchbox').show(0);</script>
        </div>
      </div>

    <div class="document">
      <div class="documentwrapper">
        <div class="bodywrapper">
          <div class="body" role="main">
            
  <div class="section" id="picking-otus-for-use-in-picrust">
<span id="otu-picking-tutorial"></span><h1>Picking OTUs for use in PICRUSt<a class="headerlink" href="#picking-otus-for-use-in-picrust" title="Permalink to this headline">¶</a></h1>
<div class="section" id="introduction">
<h2>Introduction<a class="headerlink" href="#introduction" title="Permalink to this headline">¶</a></h2>
<p>This document covers how to pick OTUs from marker gene data to use with PICRUSt. To do this, you&#8217;ll use a &#8216;closed-reference&#8217; OTU picking protocol where you search sequences against the GG reference OTUs at a specified percent identity, and discard any reads that don&#8217;t hit that reference collection.</p>
<p><strong>Note, that you can also use an &#8216;open-reference&#8217; picked OTU table from QIIME</strong> (see below: <a class="reference internal" href="#using-an-open-reference-otu-table-from-qiime">Using an open reference OTU table from QIIME</a>).</p>
<p>The newest available reference collection can be found here:</p>
<blockquote>
<div><ul class="simple">
<li>gg_13_5_otus.tar.gz (<a class="reference external" href="ftp://greengenes.microbio.me/greengenes_release/gg_13_5/gg_13_5_otus.tar.gz">download</a> )</li>
</ul>
</div></blockquote>
<p>This tutorial assumes that you have QIIME installed. See the <a class="reference external" href="http://www.qiime.org">QIIME website</a> for details on how to use QIIME. The quickest way to get started with QIIME is working on the Amazon Web Services cloud, and you can find instructions for <a class="reference external" href="http://qiime.org/tutorials/working_with_aws.html">using QIIME on the cloud here</a>.</p>
</div>
<div class="section" id="picking-closed-reference-otus-with-qiime">
<h2>Picking closed reference OTUs with QIIME<a class="headerlink" href="#picking-closed-reference-otus-with-qiime" title="Permalink to this headline">¶</a></h2>
<p>To pick &#8216;closed reference&#8217; OTUs with QIIME for use in PICRUSt, you should begin with a demuliplexed fasta file in QIIME format, and the GG reference collection (see download link above).</p>
<p>The demuliplexed fasta file in QIIME format is a standard multi-line fasta file, where sequence identifiers are of the form <code class="docutils literal"><span class="pre">sampleID_seqID</span></code>. In these sequence identifiers, <code class="docutils literal"><span class="pre">sampleID</span></code> indicates the name of the sample, and <code class="docutils literal"><span class="pre">seqID</span></code> is a unique (with respect to the current file) sequence identifier. The <code class="docutils literal"><span class="pre">seqID</span></code> values are often just assigned as ascending integers to ensure their uniqueness. These files can be generated with the <code class="docutils literal"><span class="pre">split_libraries</span></code> * scripts in QIIME, or using other software. This file might look like:</p>
<div class="highlight-python"><div class="highlight"><pre><span></span>&gt;s1_1
ACCTTAGGATGGATTAGACCCAGA
&gt;s1_2
ACCAGGATGGACCCCTTAGACCCAGA
&gt;s2_3
AGGTCCTGGATGGACCCCTTAGACCCAGA
&gt;s1_4
ACCAGTTGATGGACCCCTTAGACCCAGA
</pre></div>
</div>
<p>In this example we have two samples (<code class="docutils literal"><span class="pre">s1</span></code> and <code class="docutils literal"><span class="pre">s2</span></code>) and four sequences, three of which are in <code class="docutils literal"><span class="pre">s1</span></code> and one of which is in <code class="docutils literal"><span class="pre">s2</span></code>. For additional details on this file, see <a class="reference external" href="http://qiime.org/documentation/file_formats.html#demultiplexed-sequences">here</a>.</p>
<p>If your demultiplexed fna file is named <code class="docutils literal"><span class="pre">seqs.fna</span></code> and is in your current working directory, and you have unzipped the GG OTUs in the current working directory, you would pick OTUs with the following commands:</p>
<div class="highlight-python"><div class="highlight"><pre><span></span>echo &quot;pick_otus:enable_rev_strand_match True&quot;  &gt;&gt; $PWD/otu_picking_params_97.txt
echo &quot;pick_otus:similarity 0.97&quot; &gt;&gt; $PWD/otu_picking_params_97.txt
pick_closed_reference_otus.py -i $PWD/seqs.fna -o $PWD/ucrC97/ -p $PWD/otu_picking_params_97.txt -r $PWD/gg_13_5_otus/rep_set/97_otus.fasta -t $PWD/gg_13_5_otus/taxonomy/97_otu_taxonomy.txt
</pre></div>
</div>
<p>This command picks OTUs and builds a <a class="reference external" href="http://www.biom-format.org">biom-formatted OTU table</a>, with OTUs assigned at 97% identity. The primary file of interest will be <code class="docutils literal"><span class="pre">ucrC97/uclust_ref_picked_otus/otu_table.biom</span></code>, which will be the OTU table that you pass to PICRUSt.</p>
<p>Alternatively, if you&#8217;d like to pick OTUs at a 90% percent identity, you could run the following command:</p>
<div class="highlight-python"><div class="highlight"><pre><span></span>echo &quot;pick_otus:enable_rev_strand_match True&quot;  &gt;&gt; otu_picking_params_90.txt
echo &quot;pick_otus:similarity 0.90&quot; &gt;&gt; otu_picking_params_90.txt
pick_closed_reference_otus.py -i $PWD/seqs.fna -o $PWD/ucrC90/ -p $PWD/otu_picking_params_90.txt -r $PWD/gg_13_5_otus/rep_set/97_otus.fasta
</pre></div>
</div>
<p>If you have parallel QIIME running in your environment, you can modify the commands to run in parallel by append the <code class="docutils literal"><span class="pre">-a</span></code> flag and the <code class="docutils literal"><span class="pre">-O</span></code> parameter. This might looking like the following to use 8 processors to pick OTUs at 97% identity:</p>
<div class="highlight-python"><div class="highlight"><pre><span></span>echo &quot;pick_otus:enable_rev_strand_match True&quot;  &gt;&gt; $PWD/otu_picking_params_97.txt
echo &quot;pick_otus:similarity 0.97&quot; &gt;&gt; $PWD/otu_picking_params_97.txt
pick_closed_reference_otus.py -i $PWD/seqs.fna -o $PWD/ucrC97/ -p $PWD/otu_picking_params_97.txt -r $PWD/gg_13_5_otus/rep_set/97_otus.fasta -a -O 8
</pre></div>
</div>
<p>Note that you should not specify a value for <code class="docutils literal"><span class="pre">-O</span></code> that is greater than the number of processor/cores that you have available. This will result in longer run times as multiple jobs will compete for the same processor.</p>
</div>
<div class="section" id="using-an-open-reference-otu-table-from-qiime">
<h2>Using an open reference OTU table from QIIME<a class="headerlink" href="#using-an-open-reference-otu-table-from-qiime" title="Permalink to this headline">¶</a></h2>
<p>You may have already picked OTUs using an &#8216;open reference&#8217; approach and don&#8217;t want to have to re-pick OTUs just for PICRUSt. That is ok, but you have to first remove the de-novo OTUs by keeping only those OTUs that have matching greengene ids.</p>
<p>To make your open reference picked OTU compatible with PICRUSt:</p>
<div class="highlight-python"><div class="highlight"><pre><span></span>filter_otus_from_otu_table.py -i otu_table.biom -o closed_otu_table.biom --negate_ids_to_exclude -e $PWD/gg_13_5_otus/rep_set/97_otus.fasta
</pre></div>
</div>
</div>
<div class="section" id="removing-samples-with-0-read-counts">
<h2>Removing samples with 0 read counts<a class="headerlink" href="#removing-samples-with-0-read-counts" title="Permalink to this headline">¶</a></h2>
<p>Samples with 0 reads mapped to OTUs will be discarded at an earlier step in most OTU picking pipeline. However, if these samples are not removed then this will cause issues for PICRUSt. You can remove samples with 0 total read counts with this QIIME command:</p>
<div class="highlight-python"><div class="highlight"><pre><span></span>filter_samples_from_otu_table.py -i closed_otu_table.biom -o closed_otu_table_filt.biom -n 1
</pre></div>
</div>
</div>
</div>


          </div>
        </div>
      </div>
      <div class="clearer"></div>
    </div>
    <div class="related" role="navigation" aria-label="related navigation">
      <h3>Navigation</h3>
      <ul>
        <li class="right" style="margin-right: 10px">
          <a href="../genindex.html" title="General Index"
             >index</a></li>
        <li class="nav-item nav-item-0"><a href="../index.html">PICRUSt 1.1.3 documentation</a> &raquo;</li> 
      </ul>
    </div>
    <div class="footer" role="contentinfo">
        &copy; Copyright 2011-2013, The PICRUSt Project.
      Created using <a href="http://sphinx-doc.org/">Sphinx</a> 1.3.1.
    </div>
  </body>
</html>