
<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN"
  "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">

<html xmlns="http://www.w3.org/1999/xhtml">
  <head>
    <meta http-equiv="X-UA-Compatible" content="IE=Edge" />
    <meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
    <title>PICRUSt Script Index &#8212; PICRUSt 1.1.4 documentation</title>
    <link rel="stylesheet" href="../_static/sphinxdoc.css" type="text/css" />
    <link rel="stylesheet" href="../_static/pygments.css" type="text/css" />
    <script type="text/javascript" id="documentation_options" data-url_root="../" src="../_static/documentation_options.js"></script>
    <script type="text/javascript" src="../_static/jquery.js"></script>
    <script type="text/javascript" src="../_static/underscore.js"></script>
    <script type="text/javascript" src="../_static/doctools.js"></script>
    <script type="text/javascript" src="../_static/language_data.js"></script>
    <link rel="index" title="Index" href="../genindex.html" />
    <link rel="search" title="Search" href="../search.html" /> 
  </head><body>
    <div class="related" role="navigation" aria-label="related navigation">
      <h3>Navigation</h3>
      <ul>
        <li class="right" style="margin-right: 10px">
          <a href="../genindex.html" title="General Index"
             accesskey="I">index</a></li>
        <li class="nav-item nav-item-0"><a href="../index.html">PICRUSt 1.1.4 documentation</a> &#187;</li> 
      </ul>
    </div>
      <div class="sphinxsidebar" role="navigation" aria-label="main navigation">
        <div class="sphinxsidebarwrapper">
  <div role="note" aria-label="source link">
    <h3>This Page</h3>
    <ul class="this-page-menu">
      <li><a href="../_sources/scripts/index.rst.txt"
            rel="nofollow">Show Source</a></li>
    </ul>
   </div>
<div id="searchbox" style="display: none" role="search">
  <h3>Quick search</h3>
    <div class="searchformwrapper">
    <form class="search" action="../search.html" method="get">
      <input type="text" name="q" />
      <input type="submit" value="Go" />
      <input type="hidden" name="check_keywords" value="yes" />
      <input type="hidden" name="area" value="default" />
    </form>
    </div>
</div>
<script type="text/javascript">$('#searchbox').show(0);</script>
        </div>
      </div>

    <div class="document">
      <div class="documentwrapper">
        <div class="bodywrapper">
          <div class="body" role="main">
            
  <div class="section" id="picrust-script-index">
<span id="index-scripts"></span><h1>PICRUSt Script Index<a class="headerlink" href="#picrust-script-index" title="Permalink to this headline">¶</a></h1>
<p>Documentation for all PICRUSt scripts can be found here. This is a web version of the information that can be obtained for each script by calling the script with the <code class="docutils literal notranslate"><span class="pre">-h</span></code> parameter.</p>
<div class="toctree-wrapper compound">
<ul>
<li class="toctree-l1"><a class="reference internal" href="ancestral_state_reconstruction.html"><em>ancestral_state_reconstruction.py</em> – Runs ancestral state reconstruction given a tree and trait table</a></li>
<li class="toctree-l1"><a class="reference internal" href="categorize_by_function.html"><em>categorize_by_function.py</em> – Collapse table data to a specified level in a hierarchy.</a></li>
<li class="toctree-l1"><a class="reference internal" href="compare_biom.html"><em>compare_biom.py</em> – Compare the accuracy of biom files (expected and observed) either by observations (default) or by samples.</a></li>
<li class="toctree-l1"><a class="reference internal" href="download_picrust_files.html"><em>download_picrust_files.py</em> – Download PICRUSt precalculated files</a></li>
<li class="toctree-l1"><a class="reference internal" href="evaluate_test_datasets.html"><em>evaluate_test_datasets.py</em> – Evaluate the accuracy of character predictions, given directories of expected vs. observed test results</a></li>
<li class="toctree-l1"><a class="reference internal" href="format_tree_and_trait_table.html"><em>format_tree_and_trait_table.py</em> – Formatting script for filtering and reformatting trees and trait tables.</a></li>
<li class="toctree-l1"><a class="reference internal" href="make_test_datasets.html"><em>make_test_datasets.py</em> – Generates test datasets for cross-validation studies of PICRUSt’s accuracy</a></li>
<li class="toctree-l1"><a class="reference internal" href="metagenome_contributions.html"><em>metagenome_contributions.py</em> – This script partitions metagenome functional contributions according to function, OTU, and sample, for a given OTU table.</a></li>
<li class="toctree-l1"><a class="reference internal" href="normalize_by_copy_number.html"><em>normalize_by_copy_number.py</em> – Normalize an OTU table by marker gene copy number</a></li>
<li class="toctree-l1"><a class="reference internal" href="parallel_predict_traits.html"><em>parallel_predict_traits.py</em> – Runs predict_traits.py in parallel</a></li>
<li class="toctree-l1"><a class="reference internal" href="pool_test_datasets.html"><em>pool_test_datasets.py</em> – Pool character predictions within a directory, given directories of expected vs. observed test results</a></li>
<li class="toctree-l1"><a class="reference internal" href="predict_metagenomes.html"><em>predict_metagenomes.py</em> – This script produces the actual metagenome functional predictions for a given OTU table.</a></li>
<li class="toctree-l1"><a class="reference internal" href="predict_traits.html"><em>predict_traits.py</em> – Given a tree and a set of known character states (observed traits and reconstructions), output predictions for unobserved character states</a></li>
<li class="toctree-l1"><a class="reference internal" href="print_picrust_config.html"><em>print_picrust_config.py</em> – Print out the PICRUSt config settings.</a></li>
<li class="toctree-l1"><a class="reference internal" href="run_genome_evaluations.html"><em>run_genome_evaluations.py</em> – Runs genome evaluations on PICRUSt.</a></li>
<li class="toctree-l1"><a class="reference internal" href="scale_metagenome.html"><em>scale_metagenome.py</em> – This script converts metagenomic relative abundance back to sequence counts, by scaling the relative abundnace of each gene in each sample in a biom file by a user-supplied sequencing depth</a></li>
<li class="toctree-l1"><a class="reference internal" href="start_parallel_jobs.html"><em>start_parallel_jobs.py</em> – Starts multiple jobs in parallel on multicore or multiprocessor systems.</a></li>
<li class="toctree-l1"><a class="reference internal" href="start_parallel_jobs_sge.html"><em>start_parallel_jobs_sge.py</em> – Starts multiple jobs in parallel on SGE/qsub based multiprocessor systems.</a></li>
<li class="toctree-l1"><a class="reference internal" href="start_parallel_jobs_torque.html"><em>start_parallel_jobs_torque.py</em> – Starts multiple jobs in parallel on Torque/qsub based multiprocessor systems.</a></li>
</ul>
</div>
</div>


          </div>
        </div>
      </div>
      <div class="clearer"></div>
    </div>
    <div class="related" role="navigation" aria-label="related navigation">
      <h3>Navigation</h3>
      <ul>
        <li class="right" style="margin-right: 10px">
          <a href="../genindex.html" title="General Index"
             >index</a></li>
        <li class="nav-item nav-item-0"><a href="../index.html">PICRUSt 1.1.4 documentation</a> &#187;</li> 
      </ul>
    </div>
    <div class="footer" role="contentinfo">
        &#169; Copyright 2011-2013, The PICRUSt Project.
      Created using <a href="http://sphinx-doc.org/">Sphinx</a> 1.8.5.
    </div>
  </body>
</html>