<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN"
  "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">


<html xmlns="http://www.w3.org/1999/xhtml">
  <head>
    <meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
    
    <title>predict_traits.py – Given a tree and a set of known character states (observed traits and reconstructions), output predictions for unobserved character states &mdash; PICRUSt 1.1.3 documentation</title>
    
    <link rel="stylesheet" href="../_static/sphinxdoc.css" type="text/css" />
    <link rel="stylesheet" href="../_static/pygments.css" type="text/css" />
    
    <script type="text/javascript">
      var DOCUMENTATION_OPTIONS = {
        URL_ROOT:    '../',
        VERSION:     '1.1.3',
        COLLAPSE_INDEX: false,
        FILE_SUFFIX: '.html',
        HAS_SOURCE:  true
      };
    </script>
    <script type="text/javascript" src="../_static/jquery.js"></script>
    <script type="text/javascript" src="../_static/underscore.js"></script>
    <script type="text/javascript" src="../_static/doctools.js"></script>
    <link rel="top" title="PICRUSt 1.1.3 documentation" href="../index.html" /> 
  </head>
  <body role="document">
    <div class="related" role="navigation" aria-label="related navigation">
      <h3>Navigation</h3>
      <ul>
        <li class="right" style="margin-right: 10px">
          <a href="../genindex.html" title="General Index"
             accesskey="I">index</a></li>
        <li class="nav-item nav-item-0"><a href="../index.html">PICRUSt 1.1.3 documentation</a> &raquo;</li> 
      </ul>
    </div>
      <div class="sphinxsidebar" role="navigation" aria-label="main navigation">
        <div class="sphinxsidebarwrapper">
  <div role="note" aria-label="source link">
    <h3>This Page</h3>
    <ul class="this-page-menu">
      <li><a href="../_sources/scripts/predict_traits.txt"
            rel="nofollow">Show Source</a></li>
    </ul>
   </div>
<div id="searchbox" style="display: none" role="search">
  <h3>Quick search</h3>
    <form class="search" action="../search.html" method="get">
      <input type="text" name="q" />
      <input type="submit" value="Go" />
      <input type="hidden" name="check_keywords" value="yes" />
      <input type="hidden" name="area" value="default" />
    </form>
    <p class="searchtip" style="font-size: 90%">
    Enter search terms or a module, class or function name.
    </p>
</div>
<script type="text/javascript">$('#searchbox').show(0);</script>
        </div>
      </div>

    <div class="document">
      <div class="documentwrapper">
        <div class="bodywrapper">
          <div class="body" role="main">
            
  <span class="target" id="predict-traits"></span><div class="section" id="predict-traits-py-given-a-tree-and-a-set-of-known-character-states-observed-traits-and-reconstructions-output-predictions-for-unobserved-character-states">
<span id="index-0"></span><h1><em>predict_traits.py</em> &#8211; Given a tree and a set of known character states (observed traits and reconstructions), output predictions for unobserved character states<a class="headerlink" href="#predict-traits-py-given-a-tree-and-a-set-of-known-character-states-observed-traits-and-reconstructions-output-predictions-for-unobserved-character-states" title="Permalink to this headline">¶</a></h1>
<p><strong>Note:</strong>
This script is only needed in the case of custom predictions (e.g. extending PICRUSt to 18S or the like). In typical use cases, say KEGG Ortholog predictions from Greengenes v13.5 OTUs, you can use the <a class="reference external" href="../picrust_precalculated_files.html">precalculated files</a> to predict functions for metagenomes and calculate NSTI scores without using this script.</p>
<p><strong>Description:</strong></p>
<p>This script produces predictions of unobserved traits given a phylogenetic tree and a table that summarizes which traits are present in which ancestral organisms.
In the most common usage, this script is used to predict which gene families are present in each OTU (Operational Taxonomic Unit; roughly equivalent to a bacterial &#8216;species&#8217;), given a tree and a set of ancestral state reconstructions.</p>
<p>The output of the script is a trait prediction file, which summarizes the predicted traits of each organism of interest (by default, this is all of the organisms that are tips in the phylogenetic tree).</p>
<p>The prediction method works as follows:</p>
<blockquote>
<div><ol class="arabic simple">
<li>For each terminal (tip) node where a prediction is to be performed, the algorithm through the reconstructed ancestral states, and finds the last node in the ancestry of our organism of interest for which a prediction is available</li>
<li>The trait for the organism is then predicted based on a branch-length weighted average of the ancestral node and it&#8217;s close relatives. (This is necessary because technical limitations involving the handling of ambiguous characters in many Ancestral State Reconstruction programs prevent the parent node of the organism from being directly reconstructed in most cases.)</li>
</ol>
<p>The exact weight function to use can be specified from the commandline (see options below).</p>
<p>In general, this approach causes the prediction to be a weighted average of the closest reconstructed ancestor, and the either reconstructed or directly observed trait value of the organism of interest&#8217;s sibling node(s).</p>
</div></blockquote>
<p><strong>Usage:</strong> <code class="file docutils literal"><span class="pre">predict_traits.py</span> <span class="pre">[options]</span></code></p>
<p><strong>Input Arguments:</strong></p>
<div class="admonition note">
<p class="first admonition-title">Note</p>
<p><strong>[REQUIRED]</strong></p>
<dl class="docutils">
<dt>-i, <cite>-</cite>-observed_trait_table</dt>
<dd>The input trait table describing directly observed traits (e.g. sequenced genomes) in tab-delimited format</dd>
<dt>-t, <cite>-</cite>-tree</dt>
<dd>The full reference tree, in Newick format</dd>
</dl>
<p><strong>[OPTIONAL]</strong></p>
<dl class="last docutils">
<dt>-o, <cite>-</cite>-output_trait_table</dt>
<dd>The output filepath for trait predictions [default: predicted_states.tsv]</dd>
<dt>-a, <cite>-</cite>-calculate_accuracy_metrics</dt>
<dd>If specified, calculate accuracy metrics (i.e. how accurate does PICRUSt expect its predictions to be?) and add to output file [default: False]</dd>
<dt><cite>-</cite>-output_accuracy_metrics_only</dt>
<dd>If specified, calculate accuracy metrics (e.g. NSTI), output them to this filepath, and do not do anything else. [default: None]</dd>
<dt>-m, <cite>-</cite>-prediction_method</dt>
<dd>Specify prediction method to use.  The recommended prediction method is set as default, so other options are primarily useful for control experiments and methods validation, not typical use.  Valid choices are:asr_and_weighting,nearest_neighbor,asr_only,weighting_only,random_neighbor.  &#8220;asr_and_weighting&#8221;(recommended): use ancestral state reconstructions plus local weighting with known tip nodes.  &#8220;nearest_neighbor&#8221;: predict the closest tip on the tree with trait information.  &#8220;random_annotated_neighbor&#8221;: predict a random tip on the tree with trait information. &#8220;asr_only&#8221;: predict the traits of the last reconstructed ancestor, without weighting. &#8220;weighting_only&#8221;: weight all genomes by distance, to the organism of interest using the specified weighting function and predict the weighted average.   [default: asr_and_weighting]</dd>
<dt>-w, <cite>-</cite>-weighting_method</dt>
<dd>Specify prediction the weighting function to use.  This only applies to prediction methods that incorporate local weighting (&#8220;asr_and_weighting&#8221; or &#8220;weighting_only&#8221;)  The recommended weighting  method is set as default, so other options are primarily useful for control experiments and methods validation, not typical use.  Valid choices are:exponential,linear,equal.  &#8220;exponential&#8221;(recommended): weight genomes as a negative exponent of distance.  That is 2^-d, where d is the tip-to-tip distance from the genome to the tip.  &#8220;linear&#8221;: weight tips as a linear function of weight, normalized to the maximum possible distance (max_d -d)/d. &#8220;equal_weights&#8221;: set all weights to a constant (ignoring branch length).   [default: exponential]</dd>
<dt>-l, <cite>-</cite>-limit_predictions_by_otu_table</dt>
<dd>Specify a valid path to a legacy QIIME OTU table to perform predictions only for tips that are listed in the OTU table (regardless of abundance)</dd>
<dt>-g, <cite>-</cite>-limit_predictions_to_organisms</dt>
<dd>Limit predictions to specific, comma-separated organims ids. (Generally only useful for lists of &lt; 10 organism ids, for example when performing leave-one-out cross-validation).</dd>
<dt>-r, <cite>-</cite>-reconstructed_trait_table</dt>
<dd>The input trait table describing reconstructed traits (from <a class="reference external" href="./ancestral_state_reconstruction.html">ancestral_state_reconstruction.py</a>) in tab-delimited format [default: None]</dd>
<dt><cite>-</cite>-confidence_format</dt>
<dd>The format for the confidence intervals from ancestral state reconstruction. Only needed if passing a reconstruction confidence file with -c or &#8211;reconstruction_confidence.  These are typically sigma values for maximum likelihood ASR  methods, but 95% confidence intervals for phylogenetic independent contrasts (e.g. from the ape R packages ace function with pic as the reconstruction method).  Valid choices are:sigma,confidence_interval. [default: sigma]</dd>
<dt>-c, <cite>-</cite>-reconstruction_confidence</dt>
<dd>The input trait table describing confidence intervals for reconstructed traits (from <a class="reference external" href="./ancestral_state_reconstruction.html">ancestral_state_reconstruction.py</a>) in tab-delimited format [default: None]</dd>
<dt><cite>-</cite>-output_precalc_file_in_biom</dt>
<dd>Instead of outputting the precalculated file in tab-delimited format (with otu ids as row ids and traits as columns) output the data in biom format (with otu as SampleIds and traits as ObservationIds) [default: False]</dd>
<dt><cite>-</cite>-no_round</dt>
<dd>Flag to set if you do not want predictions to be rounded to the nearest integer [default: False]</dd>
</dl>
</div>
<p><strong>Output:</strong></p>
<p>Output is a table (tab-delimited or .biom) of predicted character states</p>
<p>Required options with NSTI:</p>
<div class="highlight-python"><div class="highlight"><pre><span></span>predict_traits.py -a -i trait_table.tab -t reference_tree.newick -r asr_counts.tab -o predict_traits.tab
</pre></div>
</div>
<p>Limit predictions to particular tips in OTU table:</p>
<div class="highlight-python"><div class="highlight"><pre><span></span>predict_traits.py -a -i trait_table.tab -t reference_tree.newick -r asr_counts.tab -o predict_traits_limited.tab -l otu_table.tab
</pre></div>
</div>
<p>Reconstruct confidence</p>
<div class="highlight-python"><div class="highlight"><pre><span></span>predict_traits.py -a -i trait_table.tab -t reference_tree.newick -r asr_counts.tab -c asr_ci.tab -o predict_traits.tab
</pre></div>
</div>
</div>


          </div>
        </div>
      </div>
      <div class="clearer"></div>
    </div>
    <div class="related" role="navigation" aria-label="related navigation">
      <h3>Navigation</h3>
      <ul>
        <li class="right" style="margin-right: 10px">
          <a href="../genindex.html" title="General Index"
             >index</a></li>
        <li class="nav-item nav-item-0"><a href="../index.html">PICRUSt 1.1.3 documentation</a> &raquo;</li> 
      </ul>
    </div>
    <div class="footer" role="contentinfo">
        &copy; Copyright 2011-2013, The PICRUSt Project.
      Created using <a href="http://sphinx-doc.org/">Sphinx</a> 1.3.1.
    </div>
  </body>
</html>